File: core/PowxN/PowxN.java
package com.dbc;

public class PowxN {
    private double cal(double x, int n) {
        if (n == 0) return 1;
        return n % 2 == 0 ? cal(x * x, n / 2) : cal(x * x, n / 2) * x;
    }

    public double myPow(double x, int n) {
        double res = cal(x, Math.abs(n));
        return n >= 0 ? res : 1 / res;
    }
}

File: core/PowxN/PowxN.py
def myPow(x: float, n: int) -> float:
    def cal(num, count):
        if count == 0:
            return 1
        return cal(num * num, count // 2) * (1 if count % 2 == 0 else num)
    res = cal(x, abs(n))
    if n >= 0:
        return res
    else:
        return 1 / res



